soroban-token-sdk/Cargo.toml

```diff
@@ -11,7 +11,6 @@ edition = "2021"
 rust-version = "1.71"
 
 [lib]
-crate-type = ["cdylib", "rlib"]
```

Comment:
I'm pretty sure we don't need to remove the `crate-type` fields. They can stay, which will make the tests in this repository more closely match examples and the guides. Can we keep them?

Reply:
I'm .. not sure what you mean. The whole point of this PR is removing them, because they force `cdylib` mode when building natively (when `cdylib` is only even appropriate / useful for wasm).

Reply:
(And doesn't work for cross compiling)

Reply:
(I should also mention: `soroban contract build` runs `cargo rustc --crate-type cdylib` so there's not a lot of need to have a `crate-type` specified in the examples either, unless you really want `cargo build --release --target wasm32-unknown-unknown` _specifically_ to build a `.wasm` file. I got the impression we were leaning into telling users to run `soroban contract build` anyways?)

Reply:
The main reason we kept the `cdylib` still in the templates is for when folks are running traditional cargo commands so that they still get .wasm files. 

Does removing the values fix the problem, or does using the rustc command fix the problem, or both? I thought the rustc command is what fixed the problem.

Reply:
I forgot also, the way that `soroban-cli` figures out which crates it should build .wasm files for, in a workspace, is it filters the crates in the workspace by crates that have `cdylib` in their `crate-type` list. It does this so that it can trivially build all contracts in a workspace where some crates are only shared libs, or are proc-macro libs.